cluster-operator/src/main/java/io/strimzi/operator/cluster/model/VolumeUtils.java

```diff
@@ -405,4 +408,132 @@ private static String getVolumeNameHashStub(String name)   {
             throw new RuntimeException("Failed to get volume name SHA-1 hash", e);
         }
     }
+
+    /**
+     * Returns the list of Volume used for authentication of Kafka client based components
+     *
+     * @param authentication    Authentication object from CRD
+     * @param tls    Kafka Client tls object
+     * @param volumeList    List where the volumes will be added
+     * @param isOpenShift   Indicates whether we run on OpenShift or not
+     * @return List of Volume
+     */
+    protected static void getVolumesTls(KafkaClientAuthentication authentication, ClientTls tls, List<Volume> volumeList, boolean isOpenShift) {
+        getVolumesTls(authentication, tls, volumeList, isOpenShift, "oauth-certs", "", false, null);
+    }
+
+    /**
+     * Returns the list of Volume used for authentication of Kafka client based components
+     *
+     * @param authentication    Authentication object from CRD
+     * @param tls    Kafka Client tls object
+     * @param volumeList    List where the volumes will be added
+     * @param isOpenShift   Indicates whether we run on OpenShift or not
+     * @param oauthVolumeNamePrefix Prefix used for OAuth volumes
+     * @return List of Volume
+     */
+    protected static void getVolumesTls(KafkaClientAuthentication authentication, ClientTls tls, List<Volume> volumeList, boolean isOpenShift, String oauthVolumeNamePrefix) {
+        getVolumesTls(authentication, tls, volumeList, isOpenShift, oauthVolumeNamePrefix, "", false, null);
+    }
+
+    /**
+     * Returns the list of Volume for authentication of Kafka client based components
+     *
+     * @param authentication    Authentication object from CRD
+     * @param tls    Kafka Client tls object
+     * @param volumeList    List where the volumes will be added
+     * @param isOpenShift   Indicates whether we run on OpenShift or not
+     * @param oauthVolumeNamePrefix Prefix used for OAuth volumes
+     * @param volumeNamePrefix Prefix used for volume names
+     * @param createOAuthSecretVolumes   Indicates whether OAuth secret volumes will be added to the list
+     * @param alias   Alias to reference the Kafka Cluster
+     * @return List of Volume
+     */
+    protected static void getVolumesTls(KafkaClientAuthentication authentication, ClientTls tls, List<Volume> volumeList, boolean isOpenShift, String oauthVolumeNamePrefix, String volumeNamePrefix, boolean createOAuthSecretVolumes,  String alias) {
```

Comment:
But this is not TLS volume, or? It seems to do lot of other stuff. You should either give it a name which corresponds or split it into smaller methods.

Reply:
Basically this method contains the auth stuff only which is being used by the Kafka Clients, should we name it something like getVolumeForKafkaClientAuth?

Reply:
I wonder if it would be better to have more smaller methods - you would have less options and you would not mix everything into one. It is not _client authentication_ either if it does TLS encryption.